client/src/components/ItineraryList/ItineraryLegDetails.tsx

```diff
@@ -2,6 +2,25 @@ import { Leg, Mode } from '../../gql/graphql.ts';
 import { LegTime } from './LegTime.tsx';
 import { formatDistance } from '../../util/formatDistance.ts';
 import { formatDuration } from '../../util/formatDuration.ts';
+import logo from '../../static/img/stay-seated.svg';
+import React from 'react';
+
+const staySeatedIcon: (leg: Leg) => React.JSX.Element = (leg: Leg) => {
+  if (leg.interchangeFrom?.staySeated) {
+    return (
+      <img
+        alt="Stay-seated transfer"
+        title="Stay-seated transfer"
+        src={logo}
+        width="20"
+        height="20"
+        className="d-inline-block align-middle"
+      />
+    );
+  } else {
+    return <></>;
+  }
+};
```

Comment:
Perhaps split this into own component as this component has potential to expand to become too big.

Reply:
Done.